python/tskit/trees.py

```diff
@@ -5442,3 +5442,45 @@ def newick_trees(self, precision=3, breakpoints=None, Ne=1):
             "This method is no longer supported. Please use the Tree.newick"
             " method instead"
         )
+
+    def nexus(self, precision=14):
+        """
+        Returns a nexus encoding <https://en.wikipedia.org/wiki/Nexus_file>`_
+        of this tree. Trees along the sequence are listed sequentially in
+        the TREES block. The tree spanning the interval :math:`[x, y)`` is
+        given the name "tree_x_y". Spatial positions are written at the
+        specified precision.
+
+        :param int precision: The numerical precision with which branch lengths
+            and tree positions are printed.
+        :return: A nexus representation of the TreeSequence.
+        :rtype: str
+        """
+        leaf_labels = {}
+        for i, u in enumerate(self.samples()):
+            leaf_labels[u] = str(i + 1)
+
+        s = "#NEXUS\n"
+        s += "BEGIN TAXA;\n"
+        s += f"\tDIMENSIONS NTAX={self.num_samples};\n"
+        s += "\tTAXLABELS\n"
+
+        for u in self.samples():
+            s += f"\t\t{u}\n"
```

Comment:
So looks like this isn't allowed in the spec:
> Taxon names are single NEXUS words. They must not correspond to another taxon name or number; thus, 1 is not a valid name for the second taxon listed. The standard defines no limit to their length, although individual programs might impose restrictions. 

We could have ``tsk_{u}`` as the taxon name? It's not totally clear to me that they are ruling out numbers here though, it's confusing.

Reply:
Re numbers, the relevant section may be this:

> Taxa, characters, and trees possess an inherent order, determined by their order of definition. This allows them to be referred to by name (if they have been given names) or by number.
Thus, in the following matrix:
MATRIX
taxon_A 001020101
taxon_B 101001022
taxon_C 001010201;
the second taxon could be referred to as taxon_B or as 2. Characters, taxa, and trees can be
referred to by their name or number

Reply:
@saunack: I used a `TRANSLATE` block to convert to zero-based numbering, FWIW.

Reply:
I think we should do away with the TAXA block entirely. It's not required by the spec, as far as I can see. In the spec they give, for example:

```
#NEXUS
BEGIN TREES;
TREE best= (fish, (frog, (snake, mouse)) ) ;
END;
```